sdk/confidentialledger/Azure.Security.CodeTransparency/tests/CcfReceiptVerifierTests.cs

```diff
@@ -0,0 +1,231 @@
+#else
+            Assert.That(ex.Message, Contains.Substring("Error while decoding COSE message"));
+#endif
+        }
+
+        [Test]
+        public void Verification_fails_with_signature_issued_elsewhere()
+        {
+            byte[] receiptBytes = readFileBytes("sbom.descriptor.2022-11-09.receipt.2023-01-23.cbor");
+            byte[] coseSign1Bytes = readFileBytes("artifact.other.cose");
+            var pem = readFileBytes("service.2022-11.cert.pem");
+            var ex = Assert.Throws<AggregateException>(() => CcfReceiptVerifier.RunVerification(certFromPem(pem), receiptBytes, coseSign1Bytes));
+#if NETFRAMEWORK
+            Assert.True(ex.Message.Contains("One or more errors occurred"));
+#else
+            Assert.That(ex.Message, Contains.Substring("VerifyReceipt: signature invalid"));
+#endif
+        }
+
+        [Test]
+        public void Verify_separate_receipt_example2_success_Test()
+        {
+            byte[] receiptBytes = readFileBytes("sbom.descriptor.2022-11-10.receipt.2023-01-23.cbor");
+            byte[] coseSign1Bytes = readFileBytes("sbom.descriptor.2022-11-10.cose");
+            var pem = readFileBytes("service.2022-11.cert.pem");
+            CcfReceiptVerifier.RunVerification(certFromPem(pem), receiptBytes, coseSign1Bytes);
+        }
+
+        [Test]
+        public void Verify_separate_receipt_example3_success_Test()
+        {
+            byte[] receiptBytes = readFileBytes("sbom.descriptor.2022-12-10.receipt.2023-01-23.cbor");
+            byte[] coseSign1Bytes = readFileBytes("sbom.descriptor.2022-12-10.cose");
+            var pem = readFileBytes("service.2022-11.cert.pem");
+            CcfReceiptVerifier.RunVerification(certFromPem(pem), receiptBytes, coseSign1Bytes);
+        }
+
+        [Test]
+        public void Verify_embedded_receipt_example_success()
+        {
+            byte[] receiptBytes = readFileBytes("sbom.descriptor.2022-12-10.embedded.2023-02-13.cose");
+            var pem = readFileBytes("service.2022-11.cert.pem");
+            CcfReceiptVerifier.RunVerification(certFromPem(pem), receiptBytes);
+        }
+
+        [Test]
+        public void Verify_embedded_receipt_is_missing_did_issuer()
+        {
+            byte[] receiptBytes = readFileBytes("sbom.descriptor.2022-12-10.embedded.2023-02-13.cose");
+            var didDocBytes = readFileBytes("service.2022-11.did.json");
+            var didDoc = DidDocument.DeserializeDidDocument(JsonDocument.Parse(didDocBytes).RootElement);
+            var ex = Assert.Throws<AggregateException>(() => CcfReceiptVerifier.RunVerification(receiptBytes, null, didRef => didDoc));
+#if NETFRAMEWORK
+            Assert.True(ex.Message.Contains("One or more errors occurred"));
+#else
+            Assert.That(ex.Message, Contains.Substring("invalid did:web issuer: null"));
+#endif
+        }
+
+        [Test]
+        public void Verify_embedded_receipt_with_did_issuer()
+        {
+            #region Snippet:CodeTransparencyVerification
```

Comment:
We generally recommend using a dedicated test class for samples, though that's not a requirement.   If you are using actual tests, you'll need to ensure that the snippet contains only the code that a customer would write to use the library - not extraneous test code.

[Example of recommended pattern](https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/keyvault/Azure.Security.KeyVault.Keys/tests/samples)

Reply:
I did not fully understand what you meant initially but now the samples are in a single test file that is used to seed the ones in the readmes. I could not achieve 1=1 as shown in the examples and did it in a separate unit test file.